front/src/modules/pathfinding/components/IncompatibleConstraints/index.tsx

```diff
@@ -0,0 +1,247 @@
+          });
+      } else {
+        result[name].enabled = !result[name].enabled;
+      }
+
+      // if on the result, nothing is selcted, then everything is selected
+      const nbEnabledAfter = toPairs(result).filter(
+        ([, value]) => value.count > 0 && value.enabled === true
+      ).length;
+      if (nbEnabledAfter === 0) {
+        filtersName.forEach((e) => {
+          result[e].enabled = true;
+        });
+      }
+
+      return result;
+    });
+  }, []);
+
+  // Function when an item is hovered in the list
+  const onListItemHover = useCallback(
+    (itemId: string | null) => setHovered(new Set(itemId ? [itemId] : [])),
+    []
+  );
+
+  // Function to toggle the selection of an item
+  const onListItemSelected = useCallback((itemId: string) => {
+    setSelected((prev) => new Set(prev.has(itemId) ? [] : [itemId]));
+  }, []);
+
+  // On map mounted, we listen some events
+  useEffect(() => {
+    const fnMouseMove = (e: MapLayerMouseEvent) => {
+      if (map.current?.getLayer('pathfinding-incompatible-constraints')) {
+        const nearestResult = getMapMouseEventNearestFeature(e, {
+          layersId: ['pathfinding-incompatible-constraints'],
+        });
+        if (nearestResult?.feature && nearestResult?.feature.properties.ids) {
+          setHovered(new Set(JSON.parse(nearestResult.feature.properties.ids)));
+        } else {
+          setHovered(new Set([]));
+        }
+      }
+    };
+    map.current?.on('mousemove', fnMouseMove);
+
+    const fnMouseClick = (e: MapLayerMouseEvent) => {
+      if (map.current?.getLayer('pathfinding-incompatible-constraints')) {
+        const nearestResult = getMapMouseEventNearestFeature(e, {
+          layersId: ['pathfinding-incompatible-constraints'],
+        });
+        if (nearestResult?.feature && nearestResult?.feature.properties.ids) {
+          setSelected((prev) => {
+            const nextSelected = JSON.parse(nearestResult.feature.properties.ids);
+            if (isArray(nextSelected)) {
+              // if we click on the same selected value
+              // => we clear the selection
+              if (prev.size === nextSelected.length && nextSelected.every((s) => prev.has(s))) {
+                return new Set([]);
+              }
+              // select the clicked elements
+              return new Set(nextSelected as string[]);
+            }
+            return new Set([]);
+          });
+        }
+      }
+    };
+    map.current?.on('mousedown', fnMouseClick);
+
+    return () => {
+      map.current?.off('mousemove', fnMouseMove);
+      map.current?.off('mousedown', fnMouseClick);
+    };
+  }, [map]);
+
+  // When pathProperties changes
+  //  => reset state
+  useEffect(() => {
+    const data =
+      pathProperties?.geometry && pathProperties?.incompatibleConstraints
+        ? pathProperties.incompatibleConstraints
+        : undefined;
+    const nextTypes: Record<string, { count: number; enabled: boolean }> = toPairs(
```

Comment:
Replace also the type with new one created in types.ts (the one from the state)

```suggestion
    const formattedIncompatibleConstraints: NewCreatedType = toPairs(
```